src/rms_cluster_manager.erl

```diff
@@ -81,6 +83,36 @@ get_cluster_riak_config(Key) ->
 get_cluster_advanced_config(Key) ->
     rms_cluster:get_field_value(advanced_config, Key).
 
+-spec get_cluster_structure(rms_cluster:key()) ->
+    {ok, [{atom(), term()}]} | {error, term()}.
+get_cluster_structure(Key) ->
+    case get_cluster(Key) of
+        {ok, Cluster} ->
+            NullIfEmptyValue = fun(Name, PropList) ->
+                                   case iolist_to_binary(proplists:get_value(Name, PropList)) of
+                                        <<>> ->
+                                            null;
+                                        Value ->
+                                            Value
+                                   end
+                               end,
+            NodeKeys = rms_node_manager:get_node_keys(Key),
+            Nodes = lists:map(fun(NodeKey) ->
+                                  {ok, Node} = rms_node_manager:get_node(NodeKey),
+                                  [{name, iolist_to_binary(proplists:get_value(key, Node))},
+                                   {status, atom_to_binary(proplists:get_value(status, Node), latin1)},
```

Comment:
Can our JSON encoder not be trusted to do this properly for us? AFAIK we use `mochijson2:encode/1` and that certainly seems to do the right thing.